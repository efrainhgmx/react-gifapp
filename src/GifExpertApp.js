import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = () => {

    
    //const categories = ['One punch', 'Pokemon', 'Dragon Ball'];
    const [categories, setCategories] = useState(['One punch']);

   // const handleAdd = () => setCategories([...categories, 'Hunter X']);


    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories= { setCategories }/>
            <hr></hr>
            <ol>
                {
                    categories.map( category => (
                            <GifGrid 
                                key= { category }
                                category= { category }
                            />
                        )
                    )

                }
            </ol>
        </>
    );
}

export default GifExpertApp;