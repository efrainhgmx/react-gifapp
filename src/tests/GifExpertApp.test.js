import React  from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Test on <GifExpertApp />', () => {
    const wrapper = shallow( <GifExpertApp /> );

    test('Should math with the snapshoot', () => {
        expect( wrapper ).toMatchSnapshot();
    })
    
})
