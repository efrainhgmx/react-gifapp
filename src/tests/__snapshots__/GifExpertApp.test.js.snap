// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GifExpertApp /> Should math with the snapshoot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch"
      key="One punch"
    />
  </ol>
</Fragment>
`;
